src/main/java/org/killbill/billing/catalog/api/VersionedCatalog.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2014-2019 Groupon, Inc
+ * Copyright 2014-2019 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.catalog.api;
+
+import java.util.Date;
+import java.util.List;
+
+public interface VersionedCatalog {
```

Comment:
Nit: should we add a `getName()` API (since it's the same name across all versions)?

Reply:
Ack will add.